flow-typed/ka-types.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+/**
+ * This is the best way to specify an empty exact type in Flow.
+ *
+ * When a component has empty props, use this type like this:
+ *     type Props = Empty;
+ *
+ * Then, when adding props to the component:
+ *     type Props = {| myComponentProp: string |};
+ *
+ * See https://github.com/facebook/flow/issues/2977 for details.
+ */
+declare type Empty = $Shape<{||}>;
+
+// Marks all properties of an object optional.
+//
+// We use this instead of $Shape<T> because spreading $Shape<$Exact<T>> ends up
+// spreading $Exact<T> instead, losing the optionality of keys.
+// https://github.com/facebook/flow/issues/6906#issuecomment-453439922
+declare type Partial<T> = $ReadOnly<$Rest<T, {...}>>;
+
+// TODO(jeff, WEB-616): Remove this in favour of real React type for refs when
+// it is available.
+declare type RefObject<T> = {current: null | React.ElementRef<T>, ...};
```

Comment:
Haven't used this one yet, so we could drop it from WB types if we wished.

Reply:
Yet another example where TypeScript types for React are ahead of the Flow types:
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L92-L96
- https://github.com/facebook/flow/blob/master/lib/react.js#L211-L214